airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -1999,8 +2009,10 @@ def create_record_selector(
                 else None,
                 **client_side_incremental_sync,
             )
-        schema_normalization = TypeTransformer(
-            SCHEMA_TRANSFORMER_TYPE_MAPPING[model.schema_normalization]
+        schema_normalization = (
+            TypeTransformer(SCHEMA_TRANSFORMER_TYPE_MAPPING[model.schema_normalization])
+            if isinstance(model.schema_normalization, SchemaNormalizationModel)
+            else self._create_component_from_model(model.schema_normalization, config=config)  # type: ignore[arg-type] # custom normalization model expected here
```

Comment:
Presumably this `CustomNormalization` component would need to already adhere or inherit from the `TypeTransformer` interface/class?

And that's why we don't need to invoke the `TypeTransformer` constructor?